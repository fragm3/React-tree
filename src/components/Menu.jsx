import React, { Component } from 'react';

export default class Menu extends Component {
    constructor() {
        super();
        this.state = {
            showMenu: false,
            menuText: 'Select traversal'
        };
    }

    showMenu = (event) => {
        this.setState({ showMenu: true }, () => {
            document.addEventListener('click', this.closeMenu);
        });
    }

    closeMenu = () => {
        this.setState({ showMenu: false }, () => {
            document.removeEventListener('click', this.closeMenu);
        }); 
    }

    selectBFT = () => {
        this.props.bft();
        this.setState({menuText: "Breadth First Traversal"});
    }

    selectinOrderT = () => {
        this.props.inorder();
        this.setState({menuText: "Inorder Traversal"});
    }

    selectpreOderT = () => {
        this.props.bft();
        this.setState({menuText: "Preorder Traversal"});
    }

    selectpostOrderT = () => {
        this.props.postOrder();
        this.setState({menuText: "Postorder Traversal"});
    }

    render() {
        return (
            <div className="menu-container">
            <div>
                <button className="menu-button" onClick={this.showMenu}>
                    {this.state.menuText}
                    <span className="caret" />
                </button>
                {this.state.showMenu ? (
                    <div className="menu"
                        ref={(element) => {
                            this.dropdownMenu = element;
                        }}
                    >
                    <ul className="dropdown-menu">
                        <li onClick={this.selectBFT}>Breadth First Traversal</li>
                        <li onClick={this.selectpreOderT}>Preorder Traversal</li>
                        <li onClick={this.selectinOrderT}>Inorder Traversal</li>
                        <li onClick={this.selectpostOrderT }>PostOrder Traversal</li>
                    </ul>
                    </div>
                    )
                    : (
                        null
                    )
                }
            </div>
            </div>
        );
    }
}