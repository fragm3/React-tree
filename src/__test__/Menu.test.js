import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Menu from '../components/Menu.jsx';

describe('<Menu />', () => {
  const wrapper = shallow(<Menu />);
  const selectMenu = wrapper.find('button.menu-button');

  it('renders Menu without crashing', () => {
    wrapper;
  });
  it('Menu matches snapshot', () => {
    const tree = renderer.create(<Menu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('dont show menu on start', () => {
    expect(wrapper.state('showMenu')).toEqual(false);
  });
  it('menu show set traversal by default', () => {
    expect(wrapper.state('menuText')).toEqual('Select traversal');
  });
  it('on click Select traversal', () => {
    selectMenu.simulate('click');
    const text = wrapper.state('menuText');
    expect(text).toEqual('Select traversal');
  });
  it('Show menu on click', () => {
    selectMenu.simulate('click');
    const text = wrapper.state('showMenu');
    expect(text).toEqual(true);
  });
});
