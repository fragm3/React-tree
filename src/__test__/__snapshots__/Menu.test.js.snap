// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Menu matches snapshot 1`] = `
<div
  className="menu-container"
>
  <div>
    <button
      className="menu-button"
      onClick={[Function]}
    >
      Select traversal
      <span
        className="caret"
      />
    </button>
  </div>
</div>
`;
